src/check_datapackage/check.py

```diff
@@ -304,6 +306,64 @@ def _handle_S_resources_x_schema_fields_x(
     return edits
 
 
+def _handle_S_resources_x_schema_fields_x_constraints_enum(
+    parent_error: SchemaError,
+    schema_errors: list[SchemaError],
+) -> SchemaErrorEdits:
+    """Only flag errors for the relevant field type and simplify errors."""
+    edits = SchemaErrorEdits(remove=[parent_error])
+    errors_in_group = _get_errors_in_group(schema_errors, parent_error)
+
+    # Remove errors for other field types
+    if _error_not_for_field_type(parent_error):
+        edits.remove.extend(errors_in_group)
+        return edits
+
+    value_errors = _filter(
+        errors_in_group,
+        lambda error: not error.jsonpath.endswith("enum"),
+    )
+
+    # If there are only value errors, simplify them
+    if value_errors == errors_in_group:
+        edits.add.append(_get_unified_enum_values_error(parent_error, value_errors))
+
+    # Otherwise, keep only top-level enum errors
+    edits.remove.extend(value_errors)
+    return edits
+
+
+def _get_unified_enum_values_error(
+    parent_error: SchemaError,
+    value_errors: list[SchemaError],
+) -> SchemaError:
+    message = "All enum values must be the same type."
+    same_type = len(set(_map(parent_error.instance, lambda value: type(value)))) == 1
+    if same_type:
+        allowed_types = set(_map(value_errors, lambda error: str(error.schema_value)))
+        message = (
+            "Incorrect enum value type. Enum values should be "
```

Comment:
```suggestion
            "The enum value type is not correct. Enum values should be "
```

Small change to make it a bit less formal sounding (to my ears anyway)